src/flag_gems/runtime/backend/__init__.py

```diff
@@ -0,0 +1,111 @@
+import ast
+import functools
+import importlib
+import os
+import sys
+
+from ..commom_utils import AUTOGRAD, Autograd, vendors, vendors_map
+
+vendors = vendors
+AUTOGRAD = AUTOGRAD
+Autograd = Autograd
+vendor_module_name = None
+vendor_module = None
+device_name = None
+device_fn_cache = {}
+
+
+def get_codegen_result(code, result_key):
+    parsed_ast = ast.parse(code)
+    compiled_code = compile(parsed_ast, filename="<ast>", mode="exec")
+    try:
+        exec(compiled_code, globals())
+    except Exception as e:
+        RuntimeError(e)
+    return globals()[result_key]
+
+
+@functools.lru_cache(maxsize=32)
+def gen_torch_tensor_attr_res(tensor, attr_name):
+    global device_name
+    device_name = device_name or get_vendor_info()[1]
+    code = f"""
+import torch
+res = {tensor}.{attr_name}
+    """
+    return get_codegen_result(code, "res")
+
+
+def gen_torch_device_fn(api_name):
+    global device_name
+    device_name = device_name or get_vendor_info()[1]
+    if api_name in device_fn_cache:
+        return device_fn_cache[api_name]
+    code = f"""
+import torch
+fn = torch.{device_name}.{api_name}()
+"""
+    fn = get_codegen_result(code, "fn")
+    device_fn_cache[api_name] = fn
+    return fn
+
+
+def get_vendor_module(vendor_name, query=False):
+    def get_module(vendor_name):
+        current_file_path = os.path.abspath(__file__)
+        current_dir_path = os.path.dirname(current_file_path)
+        sys.path.append(current_dir_path)
```

Comment:
Adding this path to `sys.path` may increase the risk to cause name conflicts with other packages.
Is there alternative ways to do this?